src/thermo/CoverageDependentSurfPhase.cpp

```diff
@@ -0,0 +1,772 @@
+/**
+ *  @file CoverageDependentSurfPhase.cpp
+ *  Definitions for a simple thermodynamic model of a coverage-dependent 
+ *  surface phase derived from SurfPhase, assuming an ideal solution model
+ *  (see \ref thermoprops and class
+ *  \link Cantera::CoverageDependentSurfPhase CoverageDependentSurfPhase\endlink).
+ */
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#include "cantera/thermo/CoverageDependentSurfPhase.h"
+#include "cantera/thermo/SurfPhase.h"
+#include "cantera/thermo/ThermoFactory.h"
+#include "cantera/base/stringUtils.h"
+#include "cantera/base/utilities.h"
+#include "cantera/base/ctml.h"
+#include "cantera/thermo/Species.h"
+
+using namespace std;
+
+namespace Cantera
+{
+CoverageDependentSurfPhase::CoverageDependentSurfPhase(doublereal n0):
+    m_covstateNum(-1),
+    m_covstateNumlast(0),
+    m_has_polynomial_dependency(false),
+    m_has_piecewise_dependency(false),
+    m_has_interpolative_dependency(false),
+    m_has_heatcapacity_dependency(false),
+    m_has_ref_coverage(false)
+{
+    setSiteDensity(n0);
+    setNDim(2);
+}
+
+CoverageDependentSurfPhase::CoverageDependentSurfPhase(const std::string& infile, 
+                                                       const std::string& id_):
+    m_covstateNum(-1),
+    m_covstateNumlast(0),
+    m_has_polynomial_dependency(false),
+    m_has_piecewise_dependency(false),
+    m_has_interpolative_dependency(false),
+    m_has_heatcapacity_dependency(false),
+    m_has_ref_coverage(false)
```

Comment:
This can all be replaced by a call to the default constructor, e.g. `CoverageDependentSurfPhase()`